File: src/Models/db/lyda/TrackCollaborator.ts
import {User} from "./User";
import {CollaboratorType} from "./CollaboratorType";
import {Track} from "./Track";

export interface TrackCollaborator {
    collab_type?: CollaboratorType;
    track?: Track;
    user?: User;
    track_id: number;
    user_id: number;
    type: number;
    approved: boolean;
    denied: boolean;
    created_at: Date;
    updated_at: Date;
}

File: kotlin/com/example/lydaandroid/model/TrackCollaborator.kt
package com.example.lydaandroid.model

import java.util.Date

data class TrackCollaborator(
    val track_id: Int,
    val user_id: Int,
    val type: Int,
    val approved: Boolean,
    val denied: Boolean,
    val created_at: String,
    val updated_at: String,
    val collab_type: CollaboratorType? = null,
    val track: Track? = null,
    val user: User? = null
)
